vmm/src/device_manager.rs

```diff
@@ -3727,6 +3727,48 @@ impl Aml for DeviceManager {
         pci_dsdt_inner_data.push(&uid);
         let supp = aml::Name::new("SUPP".into(), &aml::ZERO);
         pci_dsdt_inner_data.push(&supp);
+
+        // _DSM (Device Specific Method), the following is the implementation in ASL.
+        /*
+        Method (_DSM, 4, NotSerialized)  // _DSM: Device-Specific Method
+        {
+              If ((Arg0 == ToUUID ("e5c937d0-3553-4d7a-9117-ea4d19c3434d") /* Device Labeling Interface */))
+              {
+                  If ((Arg2 == Zero))
+                  {
+                      Return (Buffer (One)
+                      {
+                           0x21                                             // !
+                      })
+                  }
+                  If ((Arg2 == 0x05))
+                  {
+                      Return (Zero)
+                  }
+              }
+
+              Return (Buffer (One)
+              {
+                   0x00                                             // .
+              })
+        }
+        */
+        let uuid = aml::Uuid::new("E5C937D0-3553-4D7A-9117-EA4D19C3434D".to_string());
```

Comment:
We tried to refactor the code like https://github.com/cloud-hypervisor/cloud-hypervisor/pull/2241/commits/e12ab49dc792b173f7b996110bb245ff1331df38, but it doesn't build.
@rbradford, @sboeuf , could you suggest a way to make it better?

Reply:
@michael2012z I pushed a commit into your branch to simplify. Please review and squash if it's still correct.

Reply:
Fantastic! Thank you very much for improving this. I squashed into Jongwu's original commit.